src/extension/ios/iOSPlatform.ts

```diff
@@ -219,64 +200,48 @@ export class IOSPlatform extends GeneralMobilePlatform {
-        return Promise.all<boolean | string>([
+        const [debugModeEnabled, bundleId] = await Promise.all<boolean | string>([
             this.iosDebugModeManager.getAppRemoteDebuggingSetting(
                 this.runOptions.configuration,
                 this.runOptions.productName,
             ),
             this.getBundleId(),
-        ]).then(([debugModeEnabled, bundleId]) => {
-            if (debugModeEnabled) {
-                return Promise.resolve();
-            }
-
-            // Debugging must still be enabled
-            // We enable debugging by writing to a plist file that backs a NSUserDefaults object,
-            // but that file is written to by the app on occasion. To avoid races, we shut the app
-            // down before writing to the file.
-            const childProcess = new ChildProcess();
-
-            return childProcess
-                .execToString("xcrun simctl spawn booted launchctl list")
-                .then((output: string) => {
-                    // Try to find an entry that looks like UIKitApplication:com.example.myApp[0x4f37]
-                    const regex = new RegExp(`(\\S+${bundleId}\\S+)`);
-                    const match = regex.exec(output);
-
-                    // If we don't find a match, the app must not be running and so we do not need to close it
-                    return match
-                        ? childProcess.exec(`xcrun simctl spawn booted launchctl stop ${match[1]}`)
-                        : null;
-                })
-                .then(() => {
-                    // Write to the settings file while the app is not running to avoid races
-                    return this.iosDebugModeManager.setAppRemoteDebuggingSetting(
-                        /*enable=*/ true,
-                        this.runOptions.configuration,
-                        this.runOptions.productName,
-                    );
-                })
-                .then(() => {
-                    // Relaunch the app
-                    return this.runApp();
-                });
-        });
+        ]);
+        if (debugModeEnabled) {
+            return;
+        }
+        // Debugging must still be enabled
+        // We enable debugging by writing to a plist file that backs a NSUserDefaults object,
+        // but that file is written to by the app on occasion. To avoid races, we shut the app
+        // down before writing to the file.
+        const childProcess = new ChildProcess();
+        const output = await childProcess.execToString("xcrun simctl spawn booted launchctl list");
+        // Try to find an entry that looks like UIKitApplication:com.example.myApp[0x4f37]
+        const regex = new RegExp(`(\\S+${bundleId}\\S+)`);
+        const match = regex.exec(output);
+        match ? childProcess.exec(`xcrun simctl spawn booted launchctl stop ${match[1]}`) : null;
+        await this.iosDebugModeManager.setAppRemoteDebuggingSetting(
```

Comment:
```suggestion
        // Write to the settings file while the app is not running to avoid races
        await this.iosDebugModeManager.setAppRemoteDebuggingSetting(
```